libraries/Microsoft.Bot.Configuration/BotConfigurationExtensions.cs

```diff
@@ -0,0 +1,153 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Bot.Configuration
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+
+    public static class BotConfigurationExtensions
+    {
+        /// <summary>
+        /// Returns a IEnumerable of a specific <see cref="ConnectedService"/> type contained in the bot configuration file.
+        /// </summary>
+        /// <param name="botServices">BotConfiguration's services.</param>
+        /// <param name="serviceType">The type of <see cref="ConnectedService"/> to search.</param>
+        /// <returns>List of bot's <see cref="ConnectedService"/>.</returns>
+        public static IEnumerable<ConnectedService> OfType(this List<ConnectedService> botServices, string serviceType)
```

Comment:
So I'm unclear on the value of this method. The type hierarchy of `ConnectedService` actually lends itself to strongly typed service type discovery rather than `string` based. In my code I always just do this:

```CSharp
var endpointServices = botConfiguration.Services.OfType<EndpointService>();
```

That's it, done. Not only will it filter because of the type, but now I get back strongly typed results instead of just `ConnectedService`.

Even without this, is there value in this method when you can just do:

```CSharp
botConfiguration.Services.Where(s => s.Type == serviceType)
```
And get the same functionality without any additional helper code? In fact, if anything, that's really all the implementation of this method should be IMHO.

Reply:
We decided to remove this method and replace it with the usage of:

```
botConfiguration.Services.OfType<{ServiceType}>();
```